circleci: failed tests (15678): TestReplicateRogueRemovedNode

The following test appears to have failed:

[#15678](https://circleci.com/gh/cockroachdb/cockroach/15678):

```
I160331 18:39:45.370516 storage/raft_transport.go:232  remote node 1 at 127.0.0.1:41442 closed Raft transport with error: rpc error: code = 13 desc = "transport is closing"
I160331 18:39:45.371197 storage/raft_transport.go:232  remote node 1 at 127.0.0.1:41442 closed Raft transport with error: rpc error: code = 13 desc = "transport is closing"
I160331 18:39:45.371251 storage/engine/rocksdb.go:177  closing in-memory rocksdb instance
I160331 18:39:45.371407 storage/engine/rocksdb.go:177  closing in-memory rocksdb instance
I160331 18:39:45.371498 storage/engine/rocksdb.go:177  closing in-memory rocksdb instance
--- FAIL: TestReplicateRogueRemovedNode (15.37s)
	testing.go:113: storage/client_raft_test.go:1354: condition failed to evaluate within 15s: storage/client_test.go:827: expected [16 0 5], got [16 0 0]
=== RUN   TestReplicateReAddAfterDown
I160331 18:39:45.423323 storage/engine/rocksdb.go:136  opening in memory rocksdb instance
W160331 18:39:45.424195 gossip/gossip.go:885  not connected to cluster; use --join to specify a connected node
I160331 18:39:45.425792 raft/raft.go:459  [group 1] 1 became follower at term 5
I160331 18:39:45.425837 raft/raft.go:235  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I160331 18:39:45.425851 raft/raft.go:539  [group 1] 1 is starting a new election at term 5
I160331 18:39:45.425868 raft/raft.go:472  [group 1] 1 became candidate at term 6
I160331 18:39:45.425879 raft/raft.go:521  [group 1] 1 received vote from 1 at term 6
I160331 18:39:45.425900 raft/raft.go:500  [group 1] 1 became leader at term 6
--
I160331 18:40:01.222365 stopper.go:352  draining; tasks left:
104    gossip/infostore.go:286
I160331 18:40:01.222617 stopper.go:352  draining; tasks left:
103    gossip/infostore.go:286
--- PASS: Example_rebalancing (0.12s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	47.109s
=== RUN   TestBatchBasics
I160331 18:39:20.861817 storage/engine/rocksdb.go:136  opening in memory rocksdb instance
I160331 18:39:20.863184 storage/engine/rocksdb.go:177  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I160331 18:39:20.867012 storage/engine/rocksdb.go:136  opening in memory rocksdb instance
I160331 18:39:20.867658 storage/engine/rocksdb.go:177  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN   TestBatchMerge
I160331 18:39:20.869802 storage/engine/rocksdb.go:136  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -1356,7 +1356,25 @@ func TestReplicateRogueRemovedNode(t *testing.T) {
 	// Now the tombstone on node 1 prevents it from rejoining the rogue
 	// copy of the group.
 	time.Sleep(100 * time.Millisecond)
-	mtc.waitForValues(roachpb.Key("a"), []int64{16, 0, 5})
+	util.SucceedsSoon(t, func() error {
+		actual := mtc.readIntFromEngines(roachpb.Key("a"))
+		// Normally, replica GC has not happened yet on store 2, so we
+		// expect {16, 0, 5}. However, it is possible (on a
+		// slow/overloaded machine) for the end of the ChangeReplicas
+		// transaction to be queued up inside the raft transport for long
+		// enough that it doesn't arrive until after store 2 has been
+		// restarted, so it is able to trigger an early GC on the
+		// restarted node, resulting in {16, 0, 0}.
+		// TODO(bdarnell): When #5789 is fixed, the probabilities flip and
+		// {16, 0, 0} becomes the expected case. When this happens
+		// we should just combine this check with the following one.
+		expected1 := []int64{16, 0, 5}
+		expected2 := []int64{16, 0, 0}
+		if !reflect.DeepEqual(expected1, actual) && !reflect.DeepEqual(expected2, actual) {
+			return util.Errorf("expected %v or %v, got %v", expected1, expected2, actual)
+		}
+		return nil
+	})
 
 	// Run garbage collection on node 2. The lack of an active leader
 	// lease will cause GC to do a consistent range lookup, where it

